packages/scoped-elements/src/ScopedElementsMixin.js

```diff
@@ -2,35 +2,44 @@
 import { TemplateResult } from 'lit-element';
 import { dedupeMixin } from '@open-wc/dedupe-mixin';
 import { transform } from './transform.js';
+import { defineLazyElement } from './registerElement.js';
 
 /**
  * @typedef {import('lit-html/lib/shady-render').ShadyRenderOptions} ShadyRenderOptions
  * @typedef {function(TemplateResult, Element|DocumentFragment|ShadowRoot, ShadyRenderOptions): void} RenderFunction
  */
 
 /**
- * Template cache
+ * Template caches
  *
  * @type {WeakMap<Function, Map<TemplateStringsArray, TemplateStringsArray>>}
  */
 const templateCaches = new WeakMap();
 
+/**
+ * Lambda caches
+ *
+ * @type {WeakMap<object, Map<string, function(): Promise<typeof HTMLElement>>>}
+ */
+const lambdaCaches = new WeakMap();
+
 /**
  * Transforms an array of TemplateResults or arrays into another one with resolved scoped elements
  *
  * @param {ReadonlyArray} items
  * @param {Object.<string, typeof HTMLElement>} scopedElements
  * @param {Map<TemplateStringsArray, TemplateStringsArray>} cache
+ * @param {Map<string, function(): Promise<typeof HTMLElement>>} lambdaCache
```

Comment:
Type can be (() => Promise<...>).

Reply:
nice

Reply:
Intellij marks it as invalid type syntax 🤔 
![Screenshot 2020-02-27 at 17 52 51](https://user-images.githubusercontent.com/329771/75466173-14f23d00-598a-11ea-9a67-88fb050206d6.png)

Reply:
Try wrapping it in `( )` ?

Reply:
yes, also wrapping it says Invalid type syntax. Anywaay, with the new approach is not needed.